encfs: add page (#5513)
tldr-pages/tldr: pages/common/encfs.md

# encfs

> Encrypted virtual filesystem.

> > To unmount encrypted filesystem use `fusermount -u`.

Reviewer: I´m not sure about this command description @bl-ue `Encrypted virtual filesystem` this tool is for manage or only mount?
about the `fusermount` I think we should create the `fusermount` page.

Reviewer 2: @navarroaxel #5582

Reviewer 3: I would recommend changing this to a `See also`. Perhaps something like this?
```suggestion
> Mounts or creates encrypted virtual filesystems.
> See also `fusermount`, which can unmount filesystems mounted by this command.
```

This also contains a suggestion for the top line too.

> More information: <https://github.com/vgough/encfs>.

- Initialize or mount an encrypted filesystem:

`encfs {{/path/to/cipher_dir}} {{/path/to/mount_point}}`

- Initialize an encrypted filesystem with standard settings:

`encfs --standard {{/path/to/cipher_dir}} {{/path/to/mount_point}}`

> - Run encfs in foreground instead of spawning a daemon:

Reviewer 3: Perhaps this will read better?
```suggestion
- Run encfs in the foreground instead of spawning a daemon:
```

# encfs

> Encrypted virtual filesystem.
> More information: <https://github.com/vgough/encfs>.

- Initialize or mount an encrypted filesystem:

> `encfs {{/path/to/cipher_dir}} {{/path/to/mount_point}}`

Reviewer 2: ```suggestion
`encfs {{path/to/cipher_dir}} {{path/to/mount_point}}`
```

- Initialize an encrypted filesystem with standard settings:

> `encfs --standard {{/path/to/cipher_dir}} {{/path/to/mount_point}}`

Reviewer 2: ```suggestion
`encfs --standard {{path/to/cipher_dir}} {{path/to/mount_point}}`
```

Author: The path _must_ be absolute here, should I specify it in an other way ?

Reviewer 4: Then definitely leave it as is.  We prefer relative if possible, otherwise absolute.

Reviewer 2: @michaeldel should the path in the last examplie also be absolute?

Author: @marchersimon `fusermount` can work with relative paths.

- Run encfs in foreground instead of spawning a daemon:

> `encfs -f {{/path/to/cipher_dir}} {{/path/to/mount_point}}`

Reviewer 4: Is there a long option for `-f`?  No problem if there isn't.

Author: Not that I am aware of, no mention of longer alternative in manpage either.

> `encfs -f {{/path/to/cipher_dir}} {{/path/to/mount_point}}`

Reviewer 2: ```suggestion
`encfs -f {{path/to/cipher_dir}} {{path/to/mount_point}}`
```

Reviewer: @michaeldel you only removed 1 slash `/`.

Author: @navarroaxel see https://github.com/tldr-pages/tldr/pull/5513#discussion_r602779150

Author: @navarroaxel see https://github.com/tldr-pages/tldr/pull/5513#discussion_r602778748

- Unmount an encrypted filesystem:

> `fusermount -u {{/path/to/mount_point}}`

Reviewer 4: Why `fusermount`?  Do you mean this?
```suggestion
`encfs -u {{/path/to/mount_point}}`
```

Author: Using `fusermount` is the recommended way in the [`encfs` manpage](https://linux.die.net/man/1/encfs), even though `encfs -u` indeed exists. Should I stick to using the same program even though it is not really the recommended way to use it?

Reviewer 4: Hmm, I'm really not sure.  @sbrl?

Reviewer: I this this should be in the header:
```
> For unmount filesystems use `fusermount`.
```
Something like that. But I think we don´t include examples for another commands (maybe there is 1 exception around).

Author: @navarroaxel applied it at https://github.com/tldr-pages/tldr/pull/5513/commits/5b3f78819def3bfb54af04c318ca1035eed13f78

Reviewer: :thinking: I think we should add the `fusermount` page. maybe we can merge it before this.

> `fusermount -u {{/path/to/mount_point}}`

Reviewer 2: ```suggestion
`fusermount -u {{path/to/mount_point}}`
```
